File: src/main/java/ru/citeck/idea/codeinsight/forms/components/Column.java
package ru.citeck.idea.codeinsight.forms.components;

import ecos.com.fasterxml.jackson210.annotation.JsonIgnore;

public class Column extends Container {

    public static final String TYPE = "column";

    private Integer index = 0;

    @Override
    public String getType() {
        return TYPE;
    }

    public Integer getIndex() {
        return index;
    }

    public void setIndex(Integer index) {
        this.index = index;
    }

    @Override
    public String getKey() {
        return "column";
    }

    @Override
    @JsonIgnore
    public String getLabel() {
        return super.getLabel();
    }
}

File: src/main/kotlin/ru/citeck/idea/artifacts/action/codeinsight/forms/components/Column.kt
package ru.citeck.idea.artifacts.action.codeinsight.forms.components

import ecos.com.fasterxml.jackson210.annotation.JsonIgnore

class Column : Container() {

    companion object {
        const val TYPE: String = "column"
    }

    var index: Int = 0

    override fun getType(): String {
        return TYPE
    }

    override fun getKey(): String {
        return "column"
    }

    @JsonIgnore
    override fun getLabel(): String {
        return super.getLabel()
    }
}
